### PR TITLE
Fix performance issues on event info panel

### PR DESCRIPTION
**Pull request by andyrichardson**
## About

Fix expensive VDOM trees when tokenising and rendering code blocks.

## Todo

 - [x] Use mutation in reduce bodies
 - [x] Test web worker in electron
 - [ ] Update styles to use correct class names (visual regression should not be failing)
 - [x] Update types
 - [ ] Remove "click to expand" code blocks

## Notes

 - ~Uses [`dangerouslySetInnerHTML`](https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml)~

### CODE REVIEW

File: src/panel/components/CodeHighlight.tsx [hunk 1]
```diff
@@ -1,69 +1,35 @@
...
-    );
-  }
+export const CodeHighlight: FC<{ code: string; language: PrismLanguage }> = ({
+  code,
+  language,
+}) => (
+  <StyledCodeBlock className={`language language-${language}`}>
+    <code
+      dangerouslySetInnerHTML={{
+        __html: Prism.highlight(code, Prism.languages[language], language),
```
Comment (Rigellute) [inline src/panel/components/CodeHighlight.tsx:16]: If `Prism.highlight` is itself somewhat expensive, perhaps we should memoize this?

e.g. something like

```tsx
const codeHTML = useMemo(() => Prism.highlight(code, Prism.languages[language]), [code, language])

...

<code dangerouslySetInnerHTML={{ __html: codeHTML }} />
```

**Comment by kadikraman** [comment]
Regarding performance, it's still not instant for large queries (like the introspection query), but certainly an improvement!

**Review by kadikraman — APPROVED** [review]

**Comment by andyrichardson** [comment]
Is it enough of a difference?

<details>
<summary>Before</summary>

before-large-schema

</details>

<details>
<summary>After</summary>

after-large-schema


</details>

**Comment by andyrichardson** [comment]
Prism is now running in a WebWorker so rendering is not blocked while we parse big introspection queries 🙌 

Kapture 2020-09-08 at 23 03 48

**Pull request merged**